exporters/inmemory/src/test/java/io/opentelemetry/exporters/inmemory/InMemoryMetricExporterTest.java

```diff
@@ -51,26 +52,26 @@ private static MetricData generateFakeMetric() {
   }
 
   @Test
-  public void test_getFinishedMetricItems() {
+  public void test_getFinishedMetricItems() throws ExecutionException, InterruptedException {
     List<MetricData> metrics = new ArrayList<MetricData>();
     metrics.add(generateFakeMetric());
     metrics.add(generateFakeMetric());
     metrics.add(generateFakeMetric());
 
-    assertThat(exporter.export(metrics)).isEqualTo(ResultCode.SUCCESS);
+    assertThat(exporter.export(metrics).get()).isEqualTo(ResultCode.SUCCESS);
     List<MetricData> metricItems = exporter.getFinishedMetricItems();
     assertThat(metricItems).isNotNull();
     assertThat(metricItems.size()).isEqualTo(3);
   }
```

Comment:
If an exception is thrown with the export or flush methods, it is probably unrecoverable, hence allowing the exception to propagate. These methods should return a failure status for recoverable failures.